index/index.go

```diff
@@ -337,34 +360,76 @@ func (idx *Indexer) loopIteration(
 }
 
 func (idx *Indexer) ensureIndex(
-	ctx context.Context, indexType string, docType string,
+	ctx context.Context, indexType string, docType string, lang string,
 ) (string, error) {
-	name := fmt.Sprintf("%s-%s-%s",
-		indexType, idx.name, nonAlphaNum.ReplaceAllString(docType, "_"))
+	safeDocType := nonAlphaNum.ReplaceAllString(docType, "_")
+
+	config, err := GetLanguageConfig(lang)
+	if err != nil {
+		return "", fmt.Errorf("could not get language config: %w", err)
+	}
+
+	index := fmt.Sprintf("%s-%s-%s-%s", indexType, idx.name, safeDocType, config.Name)
+
+	typeAlias := fmt.Sprintf("%s-%s-%s", indexType, idx.name, safeDocType)
+	langAlias := fmt.Sprintf("%s-%s-%s-%s", indexType, idx.name, safeDocType, config.Language)
 
-	existRes, err := idx.client.Indices.Exists([]string{name},
+	settings, err := json.Marshal(config.Settings)
+	if err != nil {
+		return "", fmt.Errorf("could not marshal index settings: %w", err)
+	}
+
+	existRes, err := idx.client.Indices.Exists([]string{index},
 		idx.client.Indices.Exists.WithContext(ctx))
+
+	defer elephantine.SafeClose(idx.logger, "index exists", existRes.Body)
```

Comment:
Close must be deferred after error checking as `existRes` is `nil` after an error and we therefore would panic on accessing `existRes.Body`.

Reply:
That was a tankevurpa on my part; I assumed that if I did it after error checking we wouldn't close the Body correctly. Didn't realize that `res` would be nil.